t/app/controller/waste_peterborough.t

```diff
@@ -701,6 +703,81 @@ FixMyStreet::override_config {
     };
 };
 
+FixMyStreet::override_config {
+    ALLOWED_COBRANDS => [ 'peterborough', 'bromley' ],
+    COBRAND_FEATURES => {
+        bartec => { peterborough => {
+            sample_data => 1,
+        } },
+        waste => {
+            peterborough => 1,
+            bromley => 1
+        },
+        waste_features => {
+            peterborough => {
+                admin_config_enabled => 1
+            }
+        }
+    },
+}, sub {
+
+    subtest 'WasteWorks configuration editing' => sub {
+    ok $mech->host('peterborough.fixmystreet.com');
+
+    subtest 'Only cobrands with feature enabled are visible' => sub {
+        $mech->log_in_ok($super->email);
+        $mech->get_ok('/admin/waste');
+        $mech->content_contains("<a href=\"waste/" . $body->id . "\">Peterborough City Council</a>");
+        $mech->content_lacks("<a href=\"waste/" . $bromley->id . "\">Bromley Council</a>");
+        $mech->log_out_ok;
+    };
+
+    subtest 'Permission required to access page' => sub {
+        $mech->log_in_ok($staff->email);
+
+        $mech->get('/admin/waste/' . $body->id);
+        is $mech->res->code, 404, 'cannot access page';
+
+        $staff->user_body_permissions->create({ body => $body, permission_type => 'wasteworks_config' });
+
+        $mech->get_ok('/admin/waste/' . $body->id);
+    };
+
+    subtest 'Submitting invalid JSON shows error' => sub {
+        is $body->get_extra_metadata('wasteworks_config'), undef;
+
+        $mech->submit_form_ok({ with_fields => { body_config => '{"foo": "bar",}' } });
+        $mech->content_contains("Not a valid JSON string: &#39;&quot;&#39; expected, at character offset 14 (before &quot;}&quot;)");
+
+        $body->discard_changes;
+        is $body->get_extra_metadata('wasteworks_config'), undef;
+    };
+
+    subtest 'Submitting valid JSON but not an object shows error' => sub {
+        is $body->get_extra_metadata('wasteworks_config'), undef;
+
+        $mech->submit_form_ok({ with_fields => { body_config => '1234' } });
+        $mech->content_contains("Config must be a JSON object literal, not array.");
```

Comment:
What is the definition of an object literal? Also, you have submitted an integer/string, but the error complains about an array.

Reply:
It's one of these `{}`, as opposed to an array `[]` or a string `""` or number `1`. JSON documents have to be arrays or objects (Although I'm glad you spotted this because I'd actually configured the `JSON` object on L53 of `Admin::Waste` incorrectly and it was letting anything through when decoding as long as the syntax was OK) so if we reach this point and it's not an object it must be an array. I've rejigged these tests to describe the correct behaviour.